File: autoparts-marketplace-fe/src/app/features/notifications/notification.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from '../../core/api.service';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class NotificationService {
  constructor(private api: ApiService) {}

  sendEmail(to: string, subject: string, body: string): Observable<any> {
    return this.api.post<any>(`/api/v1/notifications/email?to=${encodeURIComponent(to)}&subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`, {});
  }

  sendNotification(notification: any): Observable<any> {
    // notification should include: id, userId, type, message, read, timestamp
    return this.api.post<any>(`/api/v1/notifications`, notification);
  }
}

File: autoparts-marketplace-backend/src/main/java/com/autoparts/marketplace/service/NotificationService.java
package com.autoparts.marketplace.service;

import org.springframework.stereotype.Service;

@Service
public class NotificationService {
    public boolean sendEmail(String to, String subject, String body) {
        // Mock email sending logic
        System.out.println("Sending email to: " + to + " | Subject: " + subject + " | Body: " + body);
        return true;
    }
}
